### ISSUE TITLE
Explicit `run-server` needed?

### ISSUE THREAD

**Issue by brendandburns** [WebAssembly/wasi-http#17]
While there is a method for handling web requests, there's no explicit method for starting a web server.

I don't know if we want to make this part of the interface, or we want to assume that there is a configured server that wraps this WASI module, but it seemed worth mentioning/asking.

cc @lukewagner

**Comment by brendandburns** [comment]
Should there at least be a method in the world that is something like `wants_http_serving()` or something like that?

**Comment by autodidaddict** [comment]
> While there is a method for handling web requests, there's no explicit method for starting a web server.
> 
> I don't know if we want to make this part of the interface, or we want to assume that there is a configured server that wraps this WASI module, but it seemed worth mentioning/asking.
> 
> cc @lukewagner

My 2 🪙:

I typically view components as pieces that sit inside hosts, and I feel like the act of starting a server is the responsibility of the host in which a component resides. The host knows what port number to use, what CORS settings to have, what TLS termination there should be, etc.. these details are ones that I typically try and keep out of a wasm module. If a component can start its own server, it now has to be responsible for all of that configuration and networking baggage that we want the host to handle for us. 

It also makes it harder for an HTTP request handler to work in both serverless and microservice style deployments, where the former might be multiplexing or proxying requests in a way opaque to the component.

In other words, in my ideal world, a component is concerned with _what_ is inside the request and _what_ to respond with, but not with _how_ that request arrived or _how_ the response will be delivered.
